src/components/Summary.tsx

```diff
@@ -1,9 +1,164 @@
+
+  useEffect(() => {
+    const fetchData = async () => {
+      try {
+        const token = await getToken();
+        const response = await getArtistInfo(token, topArtistHref);
+        setArtistData(response); // Update state with the fetched data
+      } catch (error) {
+        console.log(error);
+      }
+    };
+
+    fetchData(); // Call fetchData function when component mounts
+  }, []);
+
+  return (
+    <div className="flex justify-end">
+      <button
+        className="text-sm md:text-lg text-white bg-gray-800 hover:bg-gray-900 focus:outline-none focus:ring-4 focus:ring-gray-300 font-medium rounded-lg p-1 dark:bg-gray-800 dark:hover:bg-gray-700 dark:focus:ring-gray-700 dark:border-gray-700"
+        onClick={() => setOpen(true)}
+      >
+        Open Summary
+      </button>
+      {open && (
+        <Transition.Root show={open} as={Fragment}>
+          <Dialog as="div" className="relative z-10" initialFocus={backButtonRef} onClose={setOpen}>
+            <div className="fixed inset-0 bg-grey-500 bg-opacity-85 transition-opacity backdrop-blur-sm" />
+
+            <div className="fixed inset-0 z-10 w-screen">
+              <div className="flex min-h-full items-start justify-center p-4 text-center sm:items-center sm:p-0">
+                <Transition.Child
+                  as={Fragment}
+                  enter="ease-out duration-300"
+                  enterFrom="opacity-0 translate-y-4 sm:translate-y-0 sm:scale-95"
+                  enterTo="opacity-100 translate-y-0 sm:scale-100"
+                  leave="ease-in duration-200"
+                  leaveFrom="opacity-100 translate-y-0 sm:scale-100"
+                  leaveTo="opacity-0 translate-y-4 sm:translate-y-0 sm:scale-95"
+                >
+                  <Dialog.Panel className="relative transform overflow-hidden rounded-lg text-left shadow-xl transition-all w-full bg-slate-700 bg-opacity-80 backdrop-blur-xl drop-shadow">
+                    <div className="px-4 pt-5 sm:p-6 sm:pb-4">
+                      <div className="sm:flex sm:items-start">
+                        <Dialog.Title
+                          as="h1"
+                          className="text-base font-semibold leading-6 border-b-2 text-slate-100 text-center"
+                        >
+                          Summary ≈ {dataRange}
+                        </Dialog.Title>
+                        <div className="mt-3 text-center sm:ml-4 sm:mt-0 sm:text-left">
+                          <Dialog.Description as="div" className="mt-2 text-sm text-slate-100">
+                            Top Artist
+                          </Dialog.Description>
+                          <div className="pb-2 border-b-2">
+                            <h3 className="font-bold font-3xl mt-2 text-slate-100">
+                              {sortedArtists.entries().next().value[0]}
+                            </h3>
+                            <Image
+                              src={artistData.images[0].url}
```

Comment:
This will cause your application to throw an error if your `getArtistInfo` function errors out as it will then return an empty object.

Reply:
Added a ternary for conditional rendering to check if artistData exists

Reply:
added a check for the Image component